portfolio/src/main/java/com/google/sps/servlets/DeletionServlet.java

```diff
@@ -0,0 +1,83 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.io.IOUtils;
+
+/** Servlet that returns a programmable number of comments */
+@WebServlet("/delete-comment")
+public class DeletionServlet extends HttpServlet {
+
+  /**
+    * POST handler for the /delete-comment route
+    */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException{
+    // get the body of the request
+    String bodyString = IOUtils.toString(request.getReader());
```

Comment:
Probably not necessary to put String in the variable name as you can see it is a string from the type.

Reply:
yeah I was having a hard time deciding on this. I named the DeletionRequestBody object on line 43 'body' instead, only because that DeletionRequestBody object would be used later, whereas 'bodyString' is not referenced after line 44. I thought it made more sense therefore to reserve the name 'body' for this DeletionRequestBody object.

Reply:
What about deletionRequestJson ?

Also, what happens if the request isn't structured correctly? What is the expected behaviour?

Reply:
if the request is structured incorrectly gson.fromJson() will throw a JsonSyntaxException (https://www.javadoc.io/doc/com.google.code.gson/gson/2.8.5/com/google/gson/Gson.html#fromJson-java.lang.String-java.lang.Class-). At the moment this will just cause the servlet to quit and the client will get no response. It might make sense for me to catch this exception and return the correct http status code. I'll implement this change now

Reply:
I've implemented improved error handling for malformed json now https://github.com/thomasogara/step-repository/pull/34/commits/652da14ff7ba467c38647e59b106b3095cafe753 and renamed bodyString, I think deletionRequestJSON definitely works better for this case https://github.com/thomasogara/step-repository/pull/34/commits/06b8659b16b658e6a078a5a9ce8f6de0654f3afc